src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -148,6 +150,42 @@ public void visitJavadocToken(DetailNode ast) {
     }
 
     /**
+     * Finds if inheritDoc is placed properly in java doc.
+     * @param ast Javadoc root node.
+     * @return true if inheritDoc is valid or false.
+     */
+    private static boolean isValidInheritDoc(DetailNode ast) {
+        final Set<Integer> allowedTypes = new HashSet<>();
+        allowedTypes.add(JavadocTokenTypes.NEWLINE);
+        allowedTypes.add(JavadocTokenTypes.LEADING_ASTERISK);
+        allowedTypes.add(JavadocTokenTypes.EOF);
+        boolean result = true;
+        boolean containsInheritDoc = false;
+        for (DetailNode child : ast.getChildren()) {
+            if (child.getType() == JavadocTokenTypes.TEXT) {
+                if (!child.getText().trim().isEmpty()) {
+                    result = false;
+                }
+            }
+            else if (child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                if (child.getChildren()[1].getType() == JavadocTokenTypes.INHERIT_DOC_LITERAL) {
+                    containsInheritDoc = true;
+                }
+                else {
+                    result = false;
+                }
+            }
+            else if (!allowedTypes.contains(child.getType())) {
+                result = false;
+            }
+        }
```

Comment:
I do not see a `break` in this loop, if there be extra text at the beginning , no need to check others tags - exit with result.